### PR TITLE
Feature/issue 91/make user info component

### LINKED ISSUE #91: Make User Info Component
This is part of https://github.com/PradyumnaKrishna/enigma-protocol/issues/37, will separate the user info into its own component.

### PR DESCRIPTION
**Pull request by Tizzz-555**
closes #91 

- Introduced UserInfo Component
- Integrated ToastMessage with UserInfo
- Updated Styling
- Refactored Chat.vue
Screenshot 2023-10-11 at 20 05 33

### CODE REVIEW

File: client/src/views/Chat.vue [hunk 1]
```diff
@@ -6,29 +6,14 @@
...
-                    <button class="btn btn-dark" @click="join_room(room)">
-                      <strong>Add</strong>
-                    </button>
-                  </div>
-                </div>
-              </form>
+              <UserInfo 
+                :user="user" 
+                :toastmsg="toastmsg" 
+                :toastType="toastType"
```
Comment (PradyumnaKrishna) [inline client/src/views/Chat.vue:12]: Let's keep the toast separate from UserInfo component.

**Review by PradyumnaKrishna — CHANGES_REQUESTED** [review]

File: client/src/views/Chat.vue (hunk 1 above)
Comment (Tizzz-555) [reply client/src/views/Chat.vue:13]: Ok, I put it back in chat.vue, let me know if everything is ok now ;)

File: client/src/components/UserInfo.vue [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  </div>
+</template>
+
+<script>
+export default {
+  name: "UserInfo",
+  props: {
+    user: String,
+    toastmsg: String,
+    toastType: String,
```
Comment (PradyumnaKrishna) [inline client/src/components/UserInfo.vue:37]: Please remove these props, we don't need them.

File: client/src/views/Chat.vue [hunk 3]
```diff
@@ -167,6 +155,10 @@ export default {
         this.$cookies.set("user", this.user);
       }
     },
+    showToastMethod(message, type) {
+    this.toastmsg = message;
+    this.toastType = type;
+  },
```
Comment (PradyumnaKrishna) [inline client/src/views/Chat.vue:161]: This method is also not required.

**Review by PradyumnaKrishna — CHANGES_REQUESTED** [review]

File: client/src/views/Chat.vue [hunk 4]
```diff
@@ -6,29 +6,14 @@
...
-                    <button class="btn btn-dark" @click="join_room(room)">
-                      <strong>Add</strong>
-                    </button>
-                  </div>
-                </div>
-              </form>
+              <UserInfo 
+                :user="user" 
+                :toastmsg="toastmsg" 
+                :toastType="toastType"
```
Comment (PradyumnaKrishna) [inline client/src/views/Chat.vue:13]: There are no such props to pass.

**Review by PradyumnaKrishna — CHANGES_REQUESTED** [review]
A minor change and this will be ready to merge. 🙂

File: client/src/views/Chat.vue (hunk 4 above)
Comment (Tizzz-555) [reply client/src/views/Chat.vue:13]: Woof, sorry my bad 😅

**Review by PradyumnaKrishna — APPROVED** [review]
Thank you @Tizzz-555.

**Pull request merged**